interface/slack.py

```diff
@@ -110,6 +110,43 @@ def send_event_notif(self, message):
             logging.error("Webhook notif failed to send due to {} error.".
                           format(se.error))
 
+    def create_private_chat(self, users: List[str]) -> str:
+        """
+        Create a private chat with the given users
+
+        :param users: the list of users to add to the private chat
+
+        :raise SlackAPIError if the slack API returned error openning the chat
+
+        :return The name of of the private chat created
+        """
+        logging.debug(f"Attempting to open a private conversation with users {users}")
+        response = self.sc.conversations_open(users = users)
+        if response['ok']:
+            logging.debug(f"Successfly opened a converation with the name {response['channel']['name']}")
+            return response['channel']['name']
+        raise SlackAPIError(response['error'])
+
+    def get_channel_id(self, channel_name: str) -> str:
+        """
+        Retrieves a channel's id given it's name
+
+        :param channel_name: The name of the channel
+
+        :raise SlackAPIError if no channels were found with the name `channel_name`
+
+        :return the slack id of the channel
+        """
+        # We strip away the "#" in case it was provided with the channel name
+        channel_name = channel_name.replace("#", "")
```

Comment:
There are two cases I've seen where we ask for channel name as input:

* from Slack commands, in which case we should require the appropriate Slack mention (#535)
* from configuration, in which case we should enforce the `#` prefix when loading the configuration - there is at least one other config option where this happens, and we don't strip prefixes it seems (at least from a quick search: https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/ubclaunchpad/rocket2%24+replace%28%22%23%22%2C+%22%22%29+OR+replace%28%27%23%27%2C+%27%27%29&patternType=literal). We don't enforce this at config load time right now though, but I think that's a better option

tldr

```suggestion
```

Reply:
Hmmm yea the reason I'm keeping it here is that I'm doing the matching on channel names on my side (as apposed to on the slack API's side) and since the call to slack's `conversations_list` returns without the `#` prefix I have to deal with it 😢 A different approach could be to force it to be without the `#` in the env variables

Reply:
Ah, in that case:

```suggestion
        # We strip away the "#" in case it was provided with the channel name,
        # such as in values returned from Slack's ``conversations_list`` API.
        channel_name = channel_name.replace("#", "")
```